apple/internal/apple_framework_import.bzl

```diff
@@ -115,6 +115,9 @@ def _apple_framework_import_impl(ctx):
         if ctx.attr.weak_sdk_frameworks:
             objc_provider_fields["weak_sdk_framework"] = depset(ctx.attr.weak_sdk_frameworks)
         objc_provider_fields["static_framework_file"] = framework_imports_set
+        providers.append(
```

Comment:
this is missing the `framework_imports` from `deps`.

I'd recommend doing something like this before  `if ctx.attr.is_dynamic`:

```
transitive_sets = []
for dep in ctx.attr.deps:
    transitive_sets.append(dep[AppleFrameworkImportInfo].framework_imports)
```

then, inside the `if ctx.attr.is_dynamic:` replace the `if filtered_framework_imports` with:

```
if filtered_framework_imports:
  transitive_sets.append(depset(filtered_framework_imports))
```

and then just after the `if:else:` block, do:
```
providers.append(AppleFrameworkImportInfo(framework_imports = depset(transitive = transitive_sets)))
```

so that will take care of both dynamic and static cases, plus all of the `apple_framework_import` dependencies in `deps`

Reply:
done